From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Philipp Hancke <philipp.hancke@googlemail.com>
Date: Thu, 8 Oct 2026 11:41:00 +0200
Subject: [PATCH 03/21] Always use two-byte VP8 picture ID

which lets SFUs rewrite those without expanding or truncating the header.
This matches what libwebrtc has been doing since
  https://webrtc.googlesource.com/src/+/eb94436b386afb7165d91724893b6085365ca1ad
---
 src/aiortc/codecs/vpx.py | 5 +----
 tests/test_vpx.py        | 6 +++---
 2 files changed, 4 insertions(+), 7 deletions(-)

diff --git a/src/aiortc/codecs/vpx.py b/src/aiortc/codecs/vpx.py
index 0e8c2a0..73d852d 100644
--- a/src/aiortc/codecs/vpx.py
+++ b/src/aiortc/codecs/vpx.py
@@ -70,10 +70,7 @@ class VpxPayloadDescriptor:
         if ext_octet:
             data = pack("!BB", (1 << 7) | octet, ext_octet)
             if self.picture_id is not None:
-                if self.picture_id < 128:
-                    data += pack("!B", self.picture_id)
-                else:
-                    data += pack("!H", (1 << 15) | self.picture_id)
+                data += pack("!H", (1 << 15) | self.picture_id)
             if self.tl0picidx is not None:
                 data += pack("!B", self.tl0picidx)
             if self.tid is not None or self.keyidx is not None:
diff --git a/tests/test_vpx.py b/tests/test_vpx.py
index 0da1973..132394b 100644
--- a/tests/test_vpx.py
+++ b/tests/test_vpx.py
@@ -45,7 +45,7 @@ class VpxPayloadDescriptorTest(TestCase):
         self.assertEqual(descr.tl0picidx, None)
         self.assertEqual(descr.tid, None)
         self.assertEqual(descr.keyidx, None)
-        self.assertEqual(bytes(descr), b"\x90\x80\x11")
+        self.assertEqual(bytes(descr), b"\x90\x80\x80\x11")
         self.assertEqual(repr(descr), "VpxPayloadDescriptor(S=1, PID=0, pic_id=17)")
 
         self.assertEqual(rest, b"")
@@ -58,7 +58,7 @@ class VpxPayloadDescriptorTest(TestCase):
         self.assertEqual(descr.tl0picidx, None)
         self.assertEqual(descr.tid, None)
         self.assertEqual(descr.keyidx, None)
-        self.assertEqual(bytes(descr), b"\x90\x80\x7f")
+        self.assertEqual(bytes(descr), b"\x90\x80\x80\x7f")
 
         self.assertEqual(rest, b"")
 
@@ -200,7 +200,7 @@ class Vp8Test(CodecTestCase):
 
         packet = self.create_packet(payload=b"\x00", pts=1)
         payloads, timestamp = encoder.pack(packet)
-        self.assertEqual(payloads, [b"\x90\x80\x00\x00"])
+        self.assertEqual(payloads, [b"\x90\x80\x80\x00\x00"])
         self.assertEqual(timestamp, 90)
 
     def test_encoder_large(self) -> None:
